extensions/http-provisioner/src/main/java/org/eclipse/dataspaceconnector/transfer/provision/http/HttpProviderProvisioner.java

```diff
@@ -0,0 +1,162 @@
+        return resourceDefinition instanceof HttpProviderResourceDefinition && dataAddressType.equals(((HttpProviderResourceDefinition) resourceDefinition).getDataAddressType());
+    }
+
+    @Override
+    public boolean canDeprovision(ProvisionedResource provisionedResource) {
+        return provisionedResource instanceof HttpProvisionedContentResource &&
+                dataAddressType.equals(((HttpProvisionedContentResource) provisionedResource).getContentDataAddress().getType());
+    }
+
+    @Override
+    public CompletableFuture<ProvisionResult> provision(HttpProviderResourceDefinition resourceDefinition, Policy policy) {
+        // TODO expose scope from PolicyEngine
+        var scopedPolicy = policy;
+
+        Request request;
+        try {
+            request = createRequest(PROVISION, resourceDefinition.getTransferProcessId(), resourceDefinition.getAssetId(), scopedPolicy);
+        } catch (JsonProcessingException e) {
+            monitor.severe("Error serializing provision request for provisioner: " + name, e);
+            return completedFuture(ProvisionResult.failure(ResponseStatus.FATAL_ERROR, "Fatal error serializing request: " + e.getMessage()));
+        }
+
+        try (var response = httpClient.newCall(request).execute()) {
+            if (response.code() == 200) {
+                return completedFuture(ProvisionResult.success(new ProvisionResponse()));   // in-process
+            } else if (response.code() >= 500 && response.code() <= 504) {
+                // retry
+                return completedFuture(ProvisionResult.failure(ResponseStatus.ERROR_RETRY, "Received error code: " + response.code()));
+            } else {
+                // fatal error
+                return completedFuture(ProvisionResult.failure(ResponseStatus.FATAL_ERROR, "Received fatal error code: " + response.code()));
+            }
+        } catch (IOException e) {
+            monitor.severe("Error invoking provisioner: " + name, e);
+            return completedFuture(ProvisionResult.failure(ResponseStatus.ERROR_RETRY, "Received error: " + e.getMessage()));
+        }
+    }
+
+    @Override
+    public CompletableFuture<DeprovisionResult> deprovision(HttpProvisionedContentResource provisionedResource, Policy policy) {
+        // TODO expose scope from PolicyEngine
+        var scopedPolicy = policy;
+
+        Request request;
+        try {
+            request = createRequest(DEPROVISION, provisionedResource.getTransferProcessId(), provisionedResource.getAssetId(), scopedPolicy);
+        } catch (JsonProcessingException e) {
+            monitor.severe("Error serializing deprovision request for provisioner: " + name, e);
+            return completedFuture(DeprovisionResult.failure(ResponseStatus.FATAL_ERROR, "Fatal error serializing request: " + e.getMessage()));
+        }
+
+        try (var response = httpClient.newCall(request).execute()) {
+            if (response.code() == 200) {
```

Comment:
nit: to also catch other success codes 
```suggestion
if (response.isSuccessful()) {
```